numpy/core/arrayprint.py

```diff
@@ -859,3 +855,18 @@ def __call__(self, x):
         for field, format_function in zip(x, self.format_functions):
             s += format_function(field) + ", "
         return (s[:-2] if 1 < self.num_fields else s[:-1]) + ")"
+
+# This function implements the repr for structured-void scalars. It is called
+# from the scalartypes.c.src code. It is placed here because it uses the
+# elementwise formatters above.
+def _void_scalar_repr(x):
+    format_functions = []
+    for field_name in x.dtype.names:
+        format_function = _get_format_function(ravel(x[field_name]),
```

Comment:
How about `format_function = repr`? I might be contradicting myself here, but I think that `repr(void) should not be reconfigurable - isn't that now the case for other scalar `repr`s?

Reply:
Sorry, I missed your above comments - you already did this, and wanted me to compare.